src/scripting/ScriptingProvider.js

```diff
@@ -257,14 +257,25 @@ module.exports = class ScriptingProvider {
         }
       })
     }
+
     if (fileUtterances) {
-      fileUtterances.forEach((fileUtt) => {
-        fileUtt.sourceTag = { filename }
-      })
+      this.fileUtterances = this._tagAndCleanupUtterances(fileUtterances, filename)
     }
     return { convos: fileConvos, utterances: fileUtterances }
   }
 
+  _tagAndCleanupUtterances (utteranceFiles, filename) {
```

Comment:
that's a strange use of the "map" function - map applies a function on each element and returns the result. you should directly assign  the result of the map function the the uttterances local var.

Reply:
I updated this. The reason why I return a new object is for avoiding side effects and being immutable. I updated to use Object.assign.

Reply:
You are right I just forgot the return